compiler/rustc_ast/src/ast.rs

```diff
@@ -3374,13 +3374,19 @@ pub struct ConstItem {
     pub expr: Option<P<Expr>>,
 }
 
+#[derive(PartialEq, Eq, Clone, Copy, Encodable, Decodable, Debug, HashStable_Generic)]
+pub enum ExternCrateKind {
+    Stable,
+    Default,
```

Comment:
If we do stick with this enum:
```suggestion
    /// extern dyn crate
    Stable,
    /// extern crate
    Default,
```